internal/http/services/owncloud/ocdav/put.go

```diff
@@ -204,6 +206,36 @@ func (s *svc) handlePutHelper(w http.ResponseWriter, r *http.Request, content io
 		w.Header().Set("X-OC-Mtime", "accepted")
 	}
 
+	// curl -X PUT https://demo.owncloud.com/remote.php/webdav/testcs.bin -u demo:demo -d '123' -v -H 'OC-Checksum: SHA1:40bd001563085fc35165329ea1ff5c5ecbdbbeef'
+
+	var cparts []string
+	// TUS Upload-Checksum header takes precedence
+	if checksum := r.Header.Get("Upload-Checksum"); checksum != "" {
+		cparts = strings.SplitN(checksum, " ", 2)
+		if len(cparts) != 2 {
+			sublog.Debug().Str("upload-checksum", checksum).Msg("invalid Upload-Checksum format, expected '[algorithm] [checksum]'")
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		// Then try owncloud header
+	} else if checksum := r.Header.Get("OC-Checksum"); checksum != "" {
```

Comment:
As a bandaid, you could implement the multi `algorithm:hash` value here, so we still have compatibility with oc10

Reply:
`OC-Checksum` uses `:` as the separator. `Upload-Checksum` uses ` `. They both get forwarded with a lowercased algorithm and a ` ` separated checksum.

The OC10 code checks if the `OC-Checksum` headir is contained in the cheksum string that is stored in the database. So supporting more than one checksum in the header is currently not necessary imo.